op-challenger2/game/fault/types/position.go

```diff
@@ -154,3 +154,17 @@ func bigMSB(x *big.Int) Depth {
 	}
 	return Depth(x.BitLen() - 1)
 }
+
+func (p Position) MoveN(depth uint64, branch uint64) Position {
+	return Position{
+		depth:        p.depth + Depth(depth),
+		indexAtDepth: new(big.Int).Add(p.IndexAtDepth(), new(big.Int).Lsh(big.NewInt(int64(branch)), uint(depth))),
+	}
+}
+
+func (p Position) MoveRightN(number uint64) Position {
```

Comment:
Maybe we can be more defensive to check that `number` doesn't exceed the maximum number of steps to move right.